tokenbucket.go

```diff
@@ -122,6 +122,33 @@ func (t *TokenBucket) Limit(ctx context.Context) (time.Duration, error) {
 	return t.Take(ctx, 1)
 }
 
+// Reset the bucket to zero state
+func (t *TokenBucket) Reset(ctx context.Context) (time.Duration, error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	if err := t.locker.Lock(ctx); err != nil {
+		return 0, err
+	}
+	defer func() {
+		if err := t.locker.Unlock(ctx); err != nil {
+			t.logger.Log(err)
+		}
+	}()
+	state, err := t.backend.State(ctx)
```

Comment:
I believe you don't need you retrieve the state first and immediately override its fields.

Reply:
ohh yeah, my bad i will remove the code. Is the other stuff fine?

Reply:
Race condition occurs while using a new TokenBucketState. Without retrieving the state and using it, error occurs. 

For Reference [Memcache CompareAndSwap](https://pkg.go.dev/github.com/bradfitz/gomemcache/memcache#Client.CompareAndSwap). It expects the item's key to be the same while other values may differ.

Reply:
You can skip that check if the state is zero.

```
diff --git a/tokenbucket.go b/tokenbucket.go
index 16fc8ec..0b499b1 100644
--- a/tokenbucket.go
+++ b/tokenbucket.go
@@ -562,7 +562,7 @@ func (t *TokenBucketMemcached) SetState(ctx context.Context, state TokenBucketSt
                        Value: b.Bytes(),
                        CasID: t.casId,
                }
-               if t.raceCheck && t.casId > 0 {
+               if t.raceCheck && t.casId > 0 && !state.isZero() {
                        err = t.cli.CompareAndSwap(item)
                } else {
                        err = t.cli.Set(item)
```

Reply:
Alternatively we may consider adding a separate `Reset()` method to the `TokenBucketStateBackend` interface. 
That way we won't need a special logic in the `SetState()` to handle a reset.
However we will need to implement it for every backend.

What do you think?

Reply:
Like this? https://github.com/mennanov/limiters/pull/91